octobot_commons/databases/database_manager.py

```diff
@@ -0,0 +1,136 @@
+
+    def get_trades_db_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(), f"{constants.TRADES_DB}{self.suffix}")
+
+    def get_symbol_db_identifier(self, exchange, symbol) -> str:
+        return self._merge_parts(self._base_folder(), exchange, f"{symbol_util.merge_symbol(symbol)}{self.suffix}")
+
+    def get_backtesting_metadata_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(ignore_backtesting_id=True), f"{constants.METADATA}{self.suffix}")
+
+    def get_backtesting_run_folder(self) -> str:
+        return self._base_folder()
+
+    def get_optimizer_runs_identifier(self) -> str:
+        return self._merge_parts(self._base_folder(ignore_backtesting_id=True))
+
+    def get_optimizer_runs_schedule_identifier(self) -> str:
+        return self._merge_parts(self.base_path, constants.OPTIMIZER,
+                                 f"{constants.OPTIMIZER_RUNS_SCHEDULE_DB}{self.suffix}")
+
+    async def generate_new_backtesting_id(self) -> int:
+        return await self._generate_new_id(is_optimizer=False)
+
+    async def generate_new_optimizer_id(self, back_list) -> int:
+        return await self._generate_new_id(back_list=back_list, is_optimizer=True)
+
+    async def _generate_new_id(self, back_list=None, is_optimizer=False):
+        back_list = back_list or []
+        max_runs = constants.MAX_OPTIMIZER_RUNS if is_optimizer else constants.MAX_BACKTESTING_RUNS
+        index = 1
+        while index < max_runs:
+            if index in back_list:
+                index += 1
+                continue
+            name_candidate = self._base_folder(optimizer_id=index) if is_optimizer\
+                else self._base_folder(backtesting_id=index)
+            if self.database_adaptor == adaptors.TinyDBAdaptor:
+                if self._exists(name_candidate):
+                    index += 1
+                else:
+                    return index
+            else:
+                raise RuntimeError(f"Unsupported database adaptor runs ({self.database_adaptor}).")
+        raise RuntimeError(f"Reached maximum number of {'optimizer' if is_optimizer else 'backtesting'} runs "
+                           f"({constants.MAX_BACKTESTING_RUNS}). Please remove some.")
+
+    async def get_optimizer_run_ids(self) -> list:
+        if self.database_adaptor == adaptors.TinyDBAdaptor:
+            optimizer_runs_path = self._merge_parts(self.base_path, constants.OPTIMIZER)
+            if os.path.exists(optimizer_runs_path):
+                return [self._parse_optimizer_id(folder.name)
+                        for folder in os.scandir(optimizer_runs_path)
+                        if os.path.isdir(folder)]
+        return []
+
+    def _parse_optimizer_id(self, identifier) -> str:
+        return identifier.split(constants.DB_SEPARATOR)[-1]
+
+    def _base_folder(self, ignore_backtesting_id=False, backtesting_id=None,
```

Comment:
I don't find it logical in terms of design that the database manager is aware of whether we are in Backtesting, Optimizer or live. It should only "manage" the database.

Reply:
right, I renamed it into `RunDatabasesIdentifier`